pkg/puller/puller.go

```diff
@@ -294,16 +300,28 @@ func (p *Puller) histSyncWorker(ctx context.Context, peer swarm.Address, bin uin
 			return
 		}
 		if s > cur {
-			p.logger.Debug("histSyncWorker syncing finished", "bin", bin, "cursor", cur, "peer_address", peer)
+			p.logger.Debug("histSyncWorker syncing finished", "bin", bin, "cursor", cur, "duration", time.Since(start), "peer_address", peer)
 			return
 		}
+
+		ctx, cancel := context.WithTimeout(ctx, recalcPeersDur)
 		top, err := p.syncer.SyncInterval(ctx, peer, bin, s, cur)
 		if err != nil {
+			cancel()
 			p.metrics.HistWorkerErrCounter.Inc()
 			loggerV2.Debug("histSyncWorker syncing interval failed", "peer_address", peer, "bin", bin, "cursor", cur, "start", s, "topmost", top, "err", err)
+			if errors.Is(err, context.DeadlineExceeded) {
+				p.logger.Debug("peer sync interval timeout, exiting", "duration", time.Since(start), "peer_address", peer, "error", err)
```

Comment:
Do we need to recalcPeers or will the signal from kademlia take care of that?

Reply:
kademlia will send the signal